src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.EasyRequest.cs

```diff
@@ -722,10 +722,22 @@ internal void SetTaskOffsetCount(Task<int> task, int offset, int count)
                 }
             }
 
-            internal void  SetRedirectUri(Uri redirectUri)
+            internal void StoreLastEffectiveUri()
             {
-                _targetUri = _requestMessage.RequestUri;
-                _requestMessage.RequestUri = redirectUri;
+                IntPtr urlCharPtr; // do not free; will point to libcurl private memory
+                CURLcode urlResult = Interop.Http.EasyGetInfoPointer(_easyHandle, Interop.Http.CURLINFO.CURLINFO_EFFECTIVE_URL, out urlCharPtr);
+                if (urlResult == CURLcode.CURLE_OK && urlCharPtr != IntPtr.Zero)
+                {
+                    string url = Marshal.PtrToStringAnsi(urlCharPtr);
+                    Uri finalUri;
+                    if (Uri.TryCreate(url, UriKind.Absolute, out finalUri))
+                    {
+                        _requestMessage.RequestUri = finalUri;
+                        return;
+                    }
+                }
```

Comment:
What happens in the "else" case of all this in release? infinite redirect loop? aborted infinite redirect loop? Redirect to the new location but RequestUri stays with the old value?  If the latter, does that run the risk of exposing cookies/creds that shouldn't be?